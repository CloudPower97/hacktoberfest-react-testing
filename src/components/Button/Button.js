import React from 'react'
import PropTypes from 'prop-types'
// import cx from 'classnames'
import './Button.scss'

const Button = ({ children }) => <button className="Button">{children}</button>

Button.propTypes = {
  children: PropTypes.node.isRequired,
  onClick: PropTypes.func,
  icon: PropTypes.node,
}

export default Button
