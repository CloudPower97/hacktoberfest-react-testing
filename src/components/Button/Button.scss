.Button {
  align-items: center;
  -webkit-appearance: none;
  background-color: #a11ec6;
  border: 1px solid transparent;
  border-color: #a11ec6;
  border-radius: 0;
  border-width: 0;
  box-shadow: none;
  box-sizing: inherit;
  color: white;
  cursor: pointer;
  display: inline-flex;
  font-family: 'Space Mono', monospace;
  font-size: 1rem;
  -webkit-font-smoothing: antialiased;
  font-weight: 400;
  height: 2.25em;
  justify-content: center;
  line-height: 1.5;
  min-width: 180px;
  padding-bottom: 1.125em;
  padding-left: 2em;
  padding-right: 2em;
  padding-top: 1.125em;
  position: relative;
  text-align: center;
  text-decoration: none;
  text-rendering: optimizeLegibility;
  text-size-adjust: 100%;
  user-select: none;
  vertical-align: top;
  white-space: nowrap;

  &:not(:disabled) {
    &:hover,
    &:focus {
      background-color: #92eaff;
      border-color: #b5b5b5;
      color: #39486f;
    }
  }

  &:disabled {
    cursor: not-allowed;
    filter: brightness(0.5);
  }
}
