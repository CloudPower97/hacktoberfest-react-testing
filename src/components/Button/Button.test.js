import React from 'react'
import { shallow } from 'enzyme'
import Button from './Button'
import { FontAwesomeIcon } from '@fortawesome/react-fontawesome'
import { faArrowRight } from '@fortawesome/free-solid-svg-icons'

describe('Button', () => {
  beforeEach(() => {
    // * By default PropTypes don't throw an error, it just console.error instead
    // global.console.error = (...args) => {
    //   const propTypeFailures = [/Failed prop type/, /Warning: Received/]
    //   if (propTypeFailures.some(p => p.test(args[0]))) {
    //     throw new Error(args[0])
    //   }
    //   global.console.error(...args)
    // }
  })

  it('should render', () => {
    shallow(<Button>Hello Abstract!</Button>)
  })

  it('should match snapshot', () => {
    expect(shallow(<Button className="hacktoberfest">Hello Abstract</Button>)).toMatchSnapshot()
  })

  // * This test fails because we never render the provided icon in the component
  it('should render the provided icon', () => {
    const icon = <FontAwesomeIcon icon={faArrowRight} />
    const wrapper = shallow(<Button icon={icon}>Hello Abstract</Button>)

    expect(wrapper.contains(icon)).toEqual(true)
  })

  // * This test fails because we do not conditionally apply other className provided as prop
  it('should have custom className if provided', () => {
    const wrapper = shallow(<Button className="hacktoberfest">Hello Abstract</Button>)

    expect(wrapper.hasClass('Button hacktoberfest')).toEqual(true)
  })

  // * This test fails because we do not spread other props to the component
  it('should accept other props', () => {
    const style = {
      color: 'red',
    }
    const wrapper = shallow(
      <Button id="test" style={style} type="submit">
        Hello Abstract
      </Button>
    )

    expect(wrapper.prop('id')).toEqual('test')
    expect(wrapper.prop('style')).toEqual(style)
    expect(wrapper.prop('type')).toEqual('submit')
  })

  // * This test fails because we do not spread other props to the component
  it('should handle onClick', () => {
    const onClickMock = jest.fn()
    const wrapper = shallow(<Button onClick={onClickMock}>Hello Abstract</Button>)

    wrapper.simulate('click')

    expect(onClickMock).toHaveBeenCalledTimes(1)

    onClickMock.mockClear()
  })
})
