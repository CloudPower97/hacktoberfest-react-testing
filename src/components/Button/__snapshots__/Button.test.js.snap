// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button should match snapshot 1`] = `
<button
  className="Button"
>
  Hello Abstract
</button>
`;
